core/server/api/invites.js

```diff
@@ -145,100 +209,17 @@ invites = {
                         common.logging.warn(error.message);
                     }
 
-                    return Promise.reject(error);
-                });
-        }
-
-        function destroyOldInvite(options) {
-            var data = options.data;
-
-            return models.Invite.findOne({email: data.invites[0].email}, _.omit(options, 'data'))
-                .then(function (invite) {
-                    if (!invite) {
-                        return Promise.resolve(options);
-                    }
-
-                    return invite.destroy(options);
-                })
-                .then(function () {
-                    return options;
-                });
-        }
-
-        function validation(options) {
-            if (!options.data.invites[0].email) {
-                return Promise.reject(new common.errors.ValidationError({message: common.i18n.t('errors.api.invites.emailIsRequired')}));
-            }
-
-            if (!options.data.invites[0].role_id) {
-                return Promise.reject(new common.errors.ValidationError({message: common.i18n.t('errors.api.invites.roleIsRequired')}));
-            }
-
-            // @TODO remove when we have a new permission unit
-            // Make sure user is allowed to add a user with this role
-            // We cannot use permissible because we don't have access to the role_id!!!
-            // Adding a permissible function to the invite model, doesn't give us much context of the invite we would like to add
-            // As we are looking forward to replace the permission system completely, we do not add a hack here
-            return models.Role.findOne({id: options.data.invites[0].role_id}).then(function (roleToInvite) {
```

Comment:
this is the only functional change in this file. Remove the custom permission handling and move to the invite model.